packages/std/src/math/decimal.rs

```diff
@@ -182,6 +182,19 @@ impl Decimal {
         Self::DECIMAL_PLACES as u32
     }
 
+    pub fn floor(&self) -> Self {
+        Self((self.0 / Self::DECIMAL_FRACTIONAL) * Self::DECIMAL_FRACTIONAL)
+    }
+
+    pub fn ceil(&self) -> Self {
+        let floor = self.floor();
+        if &floor == self {
```

Comment:
Is this `&` needed? Are we missing an PartialEq implementation with reference on the one side and owned value on the other side?

Reply:
Yeah, it's not there. I can add such as a follow up.

Reply:
That would be appreciated!